File: Capítulo_4/Dividir_para_conquistar/Exercícios/soma.js
function soma(lista) {
    if (lista.length === 1) {
        // Caso-base: apenas um elemento na lista retorna o elemento
        return lista[0];
    } else {
        // Caso-recursivo: mais de um elemento na lista, retorna o primeiro elemento mais a soma dos restantes
        return lista[0] + soma(lista.slice(1));
    }
}

let total = soma([2, 4, 6]);

console.log(`A soma total é: ${total}`);

File: Capítulo_4/Dividir_para_conquistar/Exercícios/soma.py
def soma(lista):
    # Caso-base: apenas um elemento na lista retorna o elemento
    if len(lista) == 1:
        return lista[0]
    # Caso-recursivo: mais de um elemento na lista, retorna o primeiro elemento mais a soma dos restantes
    else:
        return lista[0] + soma(lista[1:])
    
total = soma([2, 4, 6])

print(f"A soma total é: {total}")